flink-tests/src/test/java/org/apache/flink/test/scheduling/UpdateJobResourceRequirementsITCase.java

```diff
@@ -0,0 +1,245 @@
+                () ->
+                        JobResourceRequirements.newBuilder()
+                                .setParallelismForJobVertex(jobVertex1.getID(), 1, 2)
+                                .setParallelismForJobVertex(jobVertex2.getID(), 1, 2)
+                                .build(),
+                initialRunningTasks,
+                runningTasksAfterRescale,
+                NUMBER_OF_SLOTS - 2);
+    }
+
+    @Test
+    void testManualUpScalingWithDifferentSlotSharingGroups() throws Exception {
+        final int initialRunningTasks = 2;
+        final int runningTasksAfterRescale = 4;
+
+        final int initialParallelism = 1;
+        final int desiredParallelism = 2;
+
+        final JobVertex jobVertex1 = new JobVertex("Operator1");
+        jobVertex1.setParallelism(initialParallelism);
+        jobVertex1.setInvokableClass(BlockingNoOpInvokable.class);
+        final JobVertex jobVertex2 = new JobVertex("Operator2");
+        jobVertex2.setParallelism(initialParallelism);
+        jobVertex2.setInvokableClass(BlockingNoOpInvokable.class);
+
+        jobVertex1.setSlotSharingGroup(new SlotSharingGroup());
+        jobVertex2.setSlotSharingGroup(new SlotSharingGroup());
+
+        final JobGraph jobGraph = JobGraphTestUtils.streamingJobGraph(jobVertex1, jobVertex2);
+
+        runRescalingTest(
+                jobGraph,
+                () ->
+                        JobResourceRequirements.newBuilder()
+                                .setParallelismForJobVertex(
+                                        jobVertex1.getID(), initialParallelism, desiredParallelism)
+                                .setParallelismForJobVertex(
+                                        jobVertex2.getID(), initialParallelism, desiredParallelism)
+                                .build(),
+                initialRunningTasks,
+                runningTasksAfterRescale,
+                NUMBER_OF_SLOTS - jobGraph.getSlotSharingGroups().size() * desiredParallelism);
+    }
+
+    @Test
+    void testManualDownScaling() throws Exception {
+        final int initialRunningTasks = 2;
+        final int runningTasksAfterRescale = 1;
+
+        final JobVertex jobVertex = new JobVertex("Operator");
+        jobVertex.setParallelism(initialRunningTasks);
+        jobVertex.setInvokableClass(BlockingNoOpInvokable.class);
+
+        final JobGraph jobGraph = JobGraphTestUtils.streamingJobGraph(jobVertex);
+
+        runRescalingTest(
+                jobGraph,
+                () ->
+                        JobResourceRequirements.newBuilder()
+                                .setParallelismForJobVertex(jobVertex.getID(), 1, 1)
+                                .build(),
+                initialRunningTasks,
+                runningTasksAfterRescale,
+                NUMBER_OF_SLOTS - 1);
+    }
+
+    private void runRescalingTest(
+            JobGraph jobGraph,
+            Supplier<JobResourceRequirements> newJobVertexParallelismSupplier,
```

Comment:
Why is this a supplier instead of a plain `JobResourceRequirements`?

Reply:
good question 🤔 🙈